src/amo/tracking.js

```diff
@@ -133,9 +142,48 @@ export class Tracking {
       // Set a custom dimension; this allows us to tell which front-end
       // (addons-frontend vs addons-server) is being used in analytics.
       ga('set', 'dimension3', 'addons-frontend');
+
+      if (convertBoolean(_config.get('trackingSendWebVitals'))) {
+        this.log('trackingSendWebVitals is enabled');
+
+        const sendWebVitalStats = this.sendWebVitalStats.bind(this);
+        _getCLS(sendWebVitalStats);
+        _getFID(sendWebVitalStats);
+        _getLCP(sendWebVitalStats);
+      }
     }
   }
 
+  sendWebVitalStats({
+    name,
+    delta,
+    id,
+  }: {|
+    name: string,
+    delta: number,
+    id: number,
+  |}) {
+    this.log('sendWebVitalStats', { name, delta, id });
+
+    this._ga('send', 'event', {
+      eventCategory: 'Web Vitals',
+      eventAction: name,
+      // The `id` value will be unique to the current page load. When sending
+      // multiple values from the same page (e.g. for CLS), Google Analytics
+      // can compute a total by grouping on this ID (note: requires
+      // `eventLabel` to be a dimension in your report).
+      eventLabel: id,
+      // Google Analytics metrics must be integers, so the value is rounded.
+      // For CLS the value is first multiplied by 1000 for greater precision
+      // (note: increase the multiplier for greater precision if needed).
+      eventValue: Math.round(name === 'CLS' ? delta * 1000 : delta),
```

Comment:
I think we need greater precision for everything. We'll probably want to be able to track improvements (or regressions) of hundreds of milliseconds or even less.

Reply:
This is the recommended config 🤷